src/services/contract-service.js

```diff
@@ -229,13 +229,14 @@ class ContractService {
     const transactionInfo = await this.web3.eth.getTransaction(hash)
 
     // transaction not mined
-    if (transactionInfo.blockNumber === null){
+    if (transactionInfo === null || transactionInfo.blockNumber === null){
       setTimeout(() => {
         this.checkForDeploymentCompletion(hash, confirmationCallback)
       }, 1500)
     } else {
       const currentBlockNumber = await this.web3.eth.getBlockNumber()
-      const confirmations = currentBlockNumber - transactionInfo.blockNumber
+      // Math.max to prevent the -1 confirmation on Rinkeby
```

Comment:
So weird ! Why does this happen on Rinkeby only and not Mainnet ? Could be related to Infura load balancing calls across multiple nodes that may not be in sync ?

Reply:
Yeah no idea why transactionInfo.blockNumber can be for 1 value bigger than currentBlockNumber. 

Interestingly on localhost the current block number doesn't even increase if the user is not making additional transactions. So I guess there are nuances between different chains. Or maybe metamask updates `currentBlockNumber` less frequently locally.